### DISCUSSION TITLE
Font generated by instancer.instantiateVariableFont cannot be use in window

### DISCUSSION THREAD

**Discussion by moi15moi** [fonttools/fonttools#2707] [Q&A] [answered]
I am trying to instantiate a font from an variable font.
I use the default axis value to instantiate it.

I wrote this code this code and it give me this [font](https://mega.nz/file/W8d2iDIS#m4JEdhtTRWmBLAGja50LPa1LRxJ3VfBAv2Mecw1zQLc
) and this [XML](https://mega.nz/file/W1c2RayK#ITOyv-8zylv_Yktd2RA1pOpEi0E-LFOkVIF-lWIowvc
).
```py
from fontTools import ttLib
from fontTools.varLib import instancer

fontsTtLib = ttLib.TTFont(r'Akshar-VariableFont_wght.ttf')

defaultCoordinates = {}
for axis in fontsTtLib["fvar"].axes:
    defaultCoordinates[axis.axisTag] = axis.defaultValue

newFont = instancer.instantiateVariableFont(fontsTtLib, defaultCoordinates, updateFontNames=True)

newFont['name'].setName("test example", 1, 3, 1, 0x409)
newFont['name'].setName("test example", 4, 3, 1, 0x409)
newFont.save("test.ttf")
newFont.saveXML("test.xml")
```

When I installed the font in windows 10, I can see it in ``HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts``, but not in ``C:\Windows\Fonts`` (I also verify that the font wasn't hidden).
I can't see the font in program like Word, photoshop, illustrator, etc.

Does this mean instantiateVariableFont has a little problem or it mean windows has problem?

**Comment by kenmcd** [comment, marked as answer]
First, where the font file is located depends on how you installed it.
Right-click and select "Install for all users" will put it in `C:\Windows\Fonts`.
If you use the Window Font Viewer and then click the `Install` button
that will install the font in your user font folder.
Hit `Windows` key + `R` to open a Run dialog, and paste this below,
`%localappdata%\Microsoft\Windows\Fonts\`
and click `OK`
That will open a File Explorer window for your user fonts folder.
The font file is probably installed there.

Second, you have some issues with the name fields inside the font.
The style group name is "test example", 
but the "Typo" name and some of the other name fields are "Akshar-Light".
Word, LibreOffice and other Windows apps use the style group name.
So in Word and LibreOffice I see the font listed as "test example"
Other applications such as Affinity Publisher use the "Typo" names,
so the font is listed as "Akshar-Light" in that application.

So you need to fix the name fields so they are all consistent.
And make sure of where you install the font file.

**Reply by moi15moi** [reply]
Yes, I installed for all users. I simply forgot to mentionned it.

I will test what you have said.

**Reply by moi15moi** [reply]
You were right.

I remove all the possible name that could make a problem and I only added the nameID 1, 2, 3 and 4 and it works.
```py
from fontTools import ttLib
from fontTools.varLib import instancer

fontsTtLib = ttLib.TTFont(r'Akshar-VariableFont_wght.ttf')

defaultCoordinates = {}
for axis in fontsTtLib["fvar"].axes:
    defaultCoordinates[axis.axisTag] = axis.defaultValue

for instance in fontsTtLib["fvar"].instances:
    if defaultCoordinates == instance.coordinates:
        style = fontsTtLib["name"].getDebugName(instance.subfamilyNameID)

newFont = instancer.instantiateVariableFont(fontsTtLib, defaultCoordinates, updateFontNames=True)

for name in newFont['name'].names:
    # List of nameID to be remove
    if name.nameID in [1, 2, 3, 4, 6, 16, 17, 21, 22, 25]:
        newFont['name'].removeNames(name.nameID, name.platformID, name.platEncID, name.langID)

fontName = "test"
# nameID to be added
newFont['name'].setName(fontName, 1, 3, 1, 0x409)
newFont['name'].setName(style, 2, 3, 1, 0x409)
newFont['name'].setName(fontName + " " + style, 3, 3, 1, 0x409)
newFont['name'].setName(fontName, 4, 3, 1, 0x409)
# 16 is not mandatory, but in order to have a better compatibility, I think it's better to keep it
newFont['name'].setName(fontName, 16, 3, 1, 0x409)

newFont.save("test.ttf")
newFont.saveXML("test.xml")
```
